mayastor/tests/lock_lba_range.rs

```diff
@@ -0,0 +1,230 @@
+        let ctx1 = get_shareable_ctx(1, 5);
+        let ctx1_clone = Arc::clone(&ctx1);
+        reactor.send_future(async move {
+            lock_range(&mut ctx1_clone.lock().unwrap()).await;
+            trace!("Second lock succeeded");
+            let _ = lock_sender.send(());
+        });
+        reactor_poll!(100);
+
+        // First lock is held, second lock shouldn't succeed
+        assert!(lock_receiver.try_recv().is_err());
+
+        // First unlock
+        let (s, r) = unbounded::<()>();
+        let ctx_clone = Arc::clone(&ctx);
+        reactor.send_future(async move {
+            unlock_range(&mut ctx_clone.lock().unwrap()).await;
+            trace!("Unlock first lock");
+            let _ = s.send(());
+        });
+        reactor_poll!(r);
+
+        // First lock released, second lock should succeed
+        assert!(lock_receiver.try_recv().is_ok());
+
+        // Second unlock
+        let (s, r) = unbounded::<()>();
+        let ctx1_clone = Arc::clone(&ctx1);
+        reactor.send_future(async move {
+            unlock_range(&mut ctx1_clone.lock().unwrap()).await;
+            trace!("Unlock second lock");
+            let _ = s.send(());
+        });
+        reactor_poll!(r);
+    }
+    test_fini();
+}
+
+#[test]
+// Test locking a block range and then issuing a front-end I/O to an overlapping
+// range.
+// TODO: Add additional test for issuing front-end I/O then taking a lock
+fn lock_then_fe_io() {
+    test_ini();
+    {
+        let reactor = Reactors::current();
+
+        // Lock range
+        let (s, r) = unbounded::<()>();
+        let ctx = get_shareable_ctx(1, 10);
+        let ctx_clone = Arc::clone(&ctx);
+        reactor.send_future(async move {
+            lock_range(&mut ctx_clone.lock().unwrap()).await;
+            trace!("*** HOLDING LOCK ***");
+            let _ = s.send(());
+        });
+        reactor_poll!(r);
+
+        // Issue front-end I/O
+        let (io_sender, io_receiver) = unbounded::<()>();
+        reactor.send_future(async move {
+            let nexus_desc = Bdev::open_by_name(&NEXUS_NAME, true).unwrap();
+            let h = nexus_desc.into_handle().unwrap();
+
+            let blk = 2;
+            let blk_size = 512;
+            let buf = DmaBuf::new(blk * blk_size, 9).unwrap();
+
+            match h.write_at((blk * blk_size) as u64, &buf).await {
+                Ok(_) => trace!("Successfully wrote to nexus"),
+                Err(e) => trace!("Failed to write to nexus: {}", e),
+            }
+
+            let _ = io_sender.send(());
+        });
+        reactor_poll!(1000);
+
+        // First lock is held, I/O should not succeed
+        assert!(io_receiver.try_recv().is_err());
+
+        // First unlock
+        let (s, r) = unbounded::<()>();
+        let ctx_clone = Arc::clone(&ctx);
+        reactor.send_future(async move {
+            unlock_range(&mut ctx_clone.lock().unwrap()).await;
+            trace!("*** RELEASE LOCK ***");
```

Comment:
capitalization/wording is different to the previous test, should it be same/similar?

Reply:
Done